types/queries.go

```diff
@@ -123,6 +125,28 @@ type AllBalancesResponse struct {
 	Amount Coins `json:"amount"`
 }
 
+type DenomMetadataQuery struct {
+	Denom string `json:"denom"`
+}
+
+type DenomMetadataResponse struct {
+	Metadata DenomMetadata `json:"metadata"`
+}
+
+type AllDenomMetadataQuery struct {
+	// optional argument
+	// default pagination will be used if this is omitted
+	Pagination *PageRequest `json:"pagination,omitempty"`
+}
+
+type AllDenomMetadataResponse struct {
+	Metadata []DenomMetadata `json:"metadata"`
+	// next_key is the key to be passed to PageRequest.key to
```

Comment:
nit: `NextKey is ...`

Reply:
Interesting, in the cosmos-sdk they seem to use the json key instead.